modules/@apostrophecms/image/ui/apos/components/AposMediaManager.vue

```diff
@@ -367,29 +351,39 @@ export default {
               draft: true
             }
           ));
-          if (!this.mounted) {
-            return;
-          }
-          this.tagList = apiResponse.choices._tags;
+          result.tagList = apiResponse.choices._tags;
         } else {
-          this.tagList = apiResponse.choices ? apiResponse.choices._tags : [];
+          result.tagList = apiResponse.choices ? apiResponse.choices._tags : [];
         }
       }
 
-      this.currentPage = apiResponse.currentPage;
-      this.totalPages = apiResponse.pages;
+      result.currentPage = apiResponse.currentPage;
+      result.totalPages = apiResponse.pages;
+      result.items = [];
       for (const image of apiResponse.results) {
-        this.items.push(image);
+        result.items.push(image);
+      }
+      return result;
+    },
+    async appendMedia({ tagList, currentPage, totalPages, items }) {
+      this.tagList = tagList;
+      this.currentPage = currentPage;
+      this.totalPages = totalPages;
+      for (const item of items) {
+        this.items.push(item);
       }
     },
     async refetchMedia(opts) {
       this.isLoading = true;
       this.currentPage = 1;
       this.items = [];
-      await this.getMedia(opts);
-      this.isLoading = false;
-      this.modified = false;
-      this.updateEditing(null);
+      try {
+        await this.debouncedGetMedia.skipDelay(opts);
+      } finally {
+        this.isLoading = false;
```

Comment:
We can still handle flags this way because it is still awaitable and waits to settle before returning, although it returns nothing of interest, not even errors. Error reporting (if we had any) would be the responsibility of `getMedia` itself.